crates/esplora/src/blocking_ext.rs

```diff
@@ -145,6 +156,7 @@ impl EsploraExt for esplora_client::BlockingClient {
         stop_gap: usize,
         parallel_requests: usize,
     ) -> Result<(TxGraph<ConfirmationTimeHeightAnchor>, BTreeMap<K, u32>), Error> {
+        assert!(stop_gap > 0, "stop_gap must be higher than 0");
```

Comment:
We should be able to handle being given a stop gap of 0 without blowing up.

Reply:
The way I think about the panics is if an app user (not dev) is able to set it to an invalid value then we should throw an error. I know wallets like [Sparrow](https://www.sparrowwallet.com/docs/faq.html#ive-restored-my-wallet-but-some-of-my-funds-are-missing) DO allow a user to manually set their stopgap when importing a wallet.  if we panic when a user set it to 0 we're giving end users a way to crash their app and that doesn't seem right. I'd prefer to only see panics on situations that can/should never happen due to end user interaction. I'd only expect a panic if to catch some serious logic error found during unit or fuzz testing.

Reply:
returning an Error now if `stop_gap == 0`

Reply:
Resolved since we are treating `stop_gap` 0 as 1.